src/main/java/seedu/address/model/expense/ExpenseDate.java

```diff
@@ -0,0 +1,49 @@
+package seedu.address.model.expense;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class ExpenseDate {
+
+
+    public static final String VALIDATION_REGEX = "^((19|2[0-9])[0-9]{2})-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])$";
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date should be in a valid format yyyy-mm-dd, and it should not be blank";
+
+    public final String expenseDate;
```

Comment:
I think expenseDate could be stored as java.time.LocalDate / java.time.LocalTime! Maybe not for this milestone but useful if we wanna sort by recency.